### DISCUSSION TITLE
On the issue of element order using the dp potential function to run lammps

### DISCUSSION THREAD

**Discussion by manbujin** [deepmodeling/deepmd-kit#2435] [Q&A] [answered]
Hello, I'm having some issues with using potential functions for lammps, hope to get some help
1.Should the order of elements in the model file of lammps be consistent with the order of elements in the training potential function?
For example, when training potential functions, my element order is Ti, Mo, Zr, C, so when I prepare the data file of lammps, must the element order also be Ti, Mo, Zr, C?
2.Should the order of the elements following“pair_coeff ” be consistent with the order of the elements in the training potential function, or should it be consistent with the order of the elements in the data file of lammps?
3.If I train the potential functions of Ti, Mo, Zr, C, but I want to use them for Mo, Zr, C, how do I write pair_coeff?
Thank you！

**Comment by chtchelkatchev** [comment]
Hi. I solved the problem using ovito-python. For my  system it was like that:
----------
from ovito.io import import_file
from ovito.io import export_file
from ovito.data import *

pipeline = import_file("POSCAR")

#def modify(frame: int, data: DataCollection):
data=pipeline.compute(0)
original = data.clone()
old_types= original.particles.particle_types

types = data.particles_.particle_types_
types.types.append(ParticleType(id=4, name='Sc'))
types.types.append(ParticleType(id=5, name='Y'))
types[old_types == 1] = 4
types[old_types == 2] = 1
types[old_types == 3] = 5

types.type_by_id_(1).name = "Gd"; types.type_by_id_(1).mass = 157.25
types.type_by_id_(2).name = "Sc"; types.type_by_id_(2).mass = 44.955
types.type_by_id_(3).name = "Y";  types.type_by_id_(3).mass = 88.906
types.type_by_id_(4).name = "Al"; types.type_by_id_(4).mass = 26.982
types.type_by_id_(5).name = "Co"; types.type_by_id_(5).mass = 58.933

export_file(data, "data", "lammps/data")

**Reply by manbujin** [reply]
Thank you very much for your answer, but I don't really understand what you mean, does it refer to the correspondence of the order of elements when the poscar file is converted to a data file?
In fact, the question I want to ask is that you need to pay attention to the order of elements when using dp training potential functions, so when using this potential function for lammps, whether the order of elements in the data file should be consistent with the training time.

**Reply by chtchelkatchev** [reply]
Using this procedure you can process lammps data files as well. Write  import_file("mydata.lammps").
About your question, yes this procedure allows me to fix the the order of elements for dp potential in lammps making the data consistent with the dp-training order of elements.

**Reply by manbujin** [reply]
Thank you very much for your answer, I will try this method

**Comment by njzjz** [comment, marked as answer]
The answer can be found in the documentation: https://docs.deepmodeling.com/projects/deepmd/en/master/third-party/lammps-command.html
Note that this is a new feature since v2.2.0.

**Reply by manbujin** [reply]
Thank you very much for your answer, I found the answer
